clientmodel/lib/src/Utopia/ClientModel.hs

```diff
@@ -3,18 +3,27 @@
 -}
 module Utopia.ClientModel where
 
-import           Control.Lens
+import           Control.Lens                                  hiding (children)
 import           Control.Monad.Fail
 import           Data.Aeson
 import           Data.Aeson.Lens
 import           Data.Aeson.Types
 import           Data.Data
 import           Data.Generics.Product
 import           Data.Generics.Sum
-import qualified Data.HashMap.Strict   as M
-import           Data.Text             hiding (foldl', reverse)
+import qualified Data.HashMap.Strict                           as M
+import           Data.Text                                     hiding (foldl',
+                                                                reverse)
 import           Data.Typeable
 import           Relude
+import           Test.QuickCheck
+import           Test.QuickCheck.Arbitrary.Generic
+import           Test.QuickCheck.Instances.Text
+import           Test.QuickCheck.Instances.UnorderedContainers
+
+-- Slightly nonsense Arbitrary instance, but one that suffices for our needs.
+instance Arbitrary Value where
+  arbitrary = fmap (toJSON :: Text -> Value) arbitrary
```

Comment:
Can all of these be moved to the `test` directory?

Reply:
So in Haskell creating orphan instances (typeclass instances not in the same file as their associated type) is frowned upon because it makes them difficult to use from multiple locations as you have to know which places to import them from.

Reply:
Urgh, that's a pain then. In that case, can I suggest we move all test code to the bottom of the file with a comment above stating that it is purely test code from that point onwards?

Reply:
I think that's just having two files in one file! :) While `Arbitrary` instances are used in test code they're not specifically test code themselves.

Reply:
Yes, but, crucially it means someone can read the code and know immediately whether or not the code they are looking at exists purely for tests (without having to resort to knowing specific conventions)

Reply:
I'm struggling to understand why that would be beneficial.

Reply:
Code readability + code file organisation. For the same reasons we typically have separate test files rather than mixing the tests in with the rest of the code. If a year from now I need to delve into this code to remind myself how the logic works and to make a change to the code, I don't want to spend extra time reading and understanding irrelevant code. I can make the choice to either start from the actual code, or start from the tests, rather than having to read functions before understanding that they are test specific functions.